### ISSUE TITLE
Prometheus integration not scraping summary metrics

### ISSUE THREAD

**Issue by Alphayeeeet** [elastic/integrations#9265]
Everything is literally mentioned in the discussion post, but no one cared about it and therefore there are no answers.

For confirmed bugs, please report:
- Version: 8.12.2
- Operating System: RHEL 8
- Discuss Forum URL: https://discuss.elastic.co/t/prometheus-integration-not-scraping-summary-metrics/352083
- Steps to Reproduce: Let Prometheus integration scrape an enpoint with summary metrics.

**Comment by lalit-satapathy** [comment]
Refer old beats issue: https://github.com/elastic/beats/issues/37376

@ishleenk17, Can you summarise what needs to happen here and if it is ready to be taken next?

**Comment by ishleenk17** [comment]
@Alphayeeeet : I am looking into it. Will get back to you soon!

**Comment by ishleenk17** [comment]
Hello,

I tried out with a service having summary metrics as below: 
<img width="752" alt="Screenshot 2024-03-12 at 8 34 40 PM" src="https://github.com/elastic/integrations/assets/102962586/12da2e13-eaef-45d5-894b-2cf0dac732c2">


I see them getting indexed in the ES:
<img width="598" alt="Screenshot 2024-03-12 at 8 34 30 PM" src="https://github.com/elastic/integrations/assets/102962586/8bdeae07-3685-4454-ac35-1c4fb315916e">

For getting the summary metrics, please ensure the use_types and rate counters are enabled (these are enabled by default)
<img width="857" alt="Screenshot 2024-03-12 at 8 35 49 PM" src="https://github.com/elastic/integrations/assets/102962586/de0ad15b-0aee-481a-b1c7-49cbd18bfd09">



Please let me know if this helps.

**Comment by Alphayeeeet** [comment]
@ishleenk17  Thank you for the reply.

I have the following prometheus metric:
![image](https://github.com/elastic/integrations/assets/52979715/4550bb36-ac38-472c-8902-fc35ceae4a2c)

...which I do net get in the event:

![image](https://github.com/elastic/integrations/assets/52979715/a6052007-0526-40ff-9abc-8186f743fe15)

Maybe it has something to do with the mappings? My events are just ingested as numbers. Still there are no separate values for each quantile (which should be available in my opinion, maybe configurable if it's optional)

My agent config is the following:
![image](https://github.com/elastic/integrations/assets/52979715/35ece901-a55e-4391-a752-636fe6c5307c)

**Comment by ishleenk17** [comment]
> Still there are no separate values for each quantile (which should be available in my opinion, maybe configurable if it's optional)

We don't give direct values of quantiles. 
Although, if you would want to check the values you can see from the ".value" field of the metric and then by opening the event you can see the quantile it corresponds to
<img width="591" alt="Screenshot 2024-03-13 at 10 15 53 AM" src="https://github.com/elastic/integrations/assets/102962586/b59497e1-0dca-48a3-9148-180b9c364de6">

**Comment by Alphayeeeet** [comment]
> > Still there are no separate values for each quantile (which should be available in my opinion, maybe configurable if it's optional)
> 
> We don't give direct values of quantiles. Although, if you would want to check the values you can see from the ".value" field of the metric and then by opening the event you can see the quantile it corresponds to <img alt="Screenshot 2024-03-13 at 10 15 53 AM" width="591" src="https://private-user-images.githubusercontent.com/102962586/312318263-b59497e1-0dca-48a3-9148-180b9c364de6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzMzYwOTYsIm5iZiI6MTcxMDMzNTc5NiwicGF0aCI6Ii8xMDI5NjI1ODYvMzEyMzE4MjYzLWI1OTQ5N2UxLTBkY2EtNDhhMy05MTQ4LTE4MGI5YzM2NGRlNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxM1QxMzE2MzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYjc3NDlhOTgzYjQyZWMzYmNhNjk2OGY2OTU4Mjc5MjM0MGZiZTk3YTRjY2Y2ODdhMjE1NWNlOTdhYWFkZjA4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.aX85z5QlrcKBmU88Yl58aIeZ9PNLs-cCQKAma4Zqo6U">

@ishleenk17 This should be a basic feature in my opinion, that I can decide, which quantile metric to use (all of them should be present in the event). Still I do not even get the '.value' field as stated out above in my screenshots. There are no metrics of '*_blobstore_file_...' present in my event. With this funcionality, the integration is not useable for our usecases...

**Comment by ishleenk17** [comment]
@Alphayeeeet : Please share the stack and integration version you are using ?

**Comment by Alphayeeeet** [comment]
@ishleenk17 
As stated out above, the stack version including the agents is 8.12.2.

Integration version:
Tried with prometheus input v0.4.0
Also tried with Prometheus 1.15.0

**Comment by Alphayeeeet** [comment]
Is it possible to file a separate feature request for a configurable option to include the quantile values as distinct fields?

**Comment by ishleenk17** [comment]
> Is it possible to open a separate feature request for a configurable option to include the quantile values as distinct fields?

Yes, we can open a ticket. Although, I would first revisit the decision of showing these quantile values as part of a single field. 
I will keep you posted on this thread.

**Comment by Alphayeeeet** [comment]
@ishleenk17 Are there any updates by now?

**Comment by canserkanuren** [comment]
Hello,
Is there any news regarding this issue ? We are having trouble monitoring our application without those metrics..

**Comment by canserkanuren** [comment]
> > Hello, Is there any news regarding this issue ? We are having trouble monitoring our application without those metrics..
> 
> There are 2 issues discussed above. Could you specify your exact requirment ?

My exact requirement would be to have summary metrics in my ElasticSearch indices by using Metricbeat & Prometheus module integration, the same way @Alphayeeeet was trying to have these metrics in his Elasticsearch cluster.

**Comment by ishleenk17** [comment]
The [summary metrics](https://prometheus.io/docs/concepts/metric_types/#summary) contains quantiles, the _count and the _sum metrics as defined by Prometheus. 
So, if a summary metric doesn't have a _sum field, it doens't get added to the metric family. We have a check for this [here](https://github.com/elastic/beats/blob/main/metricbeat/helper/prometheus/textparse.go#L673). 

Eg: The below metric would get parsed

> http_request_duration_seconds_count 5
> http_request_duration_seconds_sum 12.5
> http_request_duration_seconds{quantile="0.5"} 0.12


This metric would not get parsed.

> http_request_duration_seconds_count 5
> http_request_duration_seconds{quantile="0.5"} 0.12

Could you please check if the summary metrics that you are getting is missing the _sum value ?
As you can see [here](https://github.com/elastic/integrations/issues/9265#issuecomment-1991879340) also, summary metrics with _sum field are getting parsed. 
Also, the same would hold true for Histogram fields.

**Comment by Alphayeeeet** [comment]
@ishleenk17 Exactly that is what happens to us. The applicatuin metrics are missing the _sum field and therefore not even the quantule values are available in Elasticsearch due to the check you mentioned. Our issue is that we would like to have the distinct quantile values, when there is no average value calculated using the sum field.

**Comment by Alphayeeeet** [comment]
@ishleenk17 Are there any updates since my last comment, on how this will be adressed? Thank you

**Comment by ishleenk17** [comment]
I have apprised my team about this requirement.
Could you please let me know this is for which customer? That would help in prioritization.

cc: @lalit-satapathy

**Comment by Alphayeeeet** [comment]
@ishleenk17 We are not a customer yet.

**Comment by Alphayeeeet** [comment]
+1
